python/utils/worker.py

```diff
@@ -92,195 +92,258 @@ def run(self):
+        prev_processed_start_version = None
+        prev_processed_end_version = None
+
+        print(
+            json.dumps(
+                {
+                    "processor_name": self.processor.name(),
+                    "message": "[Parser] Connecting and making requests to GRPC endpoint",
+                }
             ),
-            (
-                "grpc.keepalive_timeout_ms",
-                self.config.indexer_grpc_http2_ping_timeout_in_secs * 1000,
-            ),
-        ]
+            flush=True,
+        )
 
-        # Connect to indexer grpc endpoint
-        with grpc.insecure_channel(
-            self.config.grpc_data_stream_endpoint, options=options
-        ) as channel:
-            print(
-                json.dumps(
-                    {
-                        "message": f"Connected to grpc data stream endpoint: {self.config.grpc_data_stream_endpoint}",
-                        "starting_version": starting_version,
-                    }
-                ),
-                flush=True,
-            )
+        # Add reconnection stuff here
+        response_stream = self.get_transaction_response_stream(starting_version)
+        print(
+            json.dumps(
+                {
+                    "processor_name": self.processor.name(),
+                    "message": f"[Parser] Successfully connected to GRPC endpoint: {self.config.grpc_data_stream_endpoint}",
+                    "starting_version": starting_version,
+                }
+            ),
+            flush=True,
+        )
 
-            batch_start_version = starting_version
-            batch_end_version = None
-            prev_processed_start_version = None
-            prev_processed_end_version = None
+        grpc_last_reconnection_time = None
+        while True:
+            perf_start_time = perf_counter()
+            transaction_batches: List[List[transaction_pb2.Transaction]] = []
+
+            # Get batches of transactions for processing
+            for _ in range(self.num_concurrent_processing_tasks):
+                response = None
+
+                try:
+                    response = next(response_stream)
+                except grpc.RpcError:
+                    # Handle GRPC reconnection
+                    if grpc_last_reconnection_time:
+                        sec_since_last_reconnection = (
+                            perf_counter() - grpc_last_reconnection_time
+                        )
+                        if (
+                            sec_since_last_reconnection
+                            < INDEXER_GRPC_MIN_SEC_BETWEEN_GRPC_RECONNECTS
+                        ):
+                            print(
+                                json.dumps(
+                                    {
+                                        "processor_name": self.processor.name(),
+                                        "message": "[Parser] Recently reconnected to GRPC stream. Will not retry",
```

Comment:
Let's standardize these logs? In rust it's `[Parser] Recently reconnected. Will not retry`

I know that it's not easy keeping these always in sync, maybe we could look into some type of constants across languages. It's not great having different logging messages for different languages as it makes debugging down the line much more difficult.